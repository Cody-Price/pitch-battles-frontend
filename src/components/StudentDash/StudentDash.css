.student-dash {
  font-family: "chronotype";
  min-height: 100vh;
  display: flex;
  align-items: center;
  position: relative;
}

.snow-backer {
  animation: snow 90s infinite linear;
  position: absolute;
  height: 6000px;
  width: 10000px;
  background: url(../Landing/images/snow/wsheet_snow_1.png);
  background-size: 384px 96px;
  z-index: 10000;
}

.student-dash-floating-backer {
  position: absolute;
  height: 1500px;
  width: 2500px;
  animation: 90s background-move infinite linear;
  background-size: cover;
  z-index: 1;
  overflow: hidden;
}

.student-dash-wrapper {
  width: 800px;
  height: 600px;
  background: url("../Game/images/char-dash-backer.png");
  background-size: cover;
  image-rendering: pixelated;
  margin: auto;
  position: relative;
  z-index: 1000;
  box-shadow: 2px 2px 2px rgba(0, 0, 0.3);
}

.student-dash-header {
  display: grid;
  grid-template-columns: 112px 1fr;

  position: relative;
}

.student-dash-avatar-backer {
  grid-row: 1/3;
  margin: 2rem 2rem 2rem 2rem;
  position: relative;
  height: 80px;
  width: 80px;
  border-radius: 50%;
  background: url(../Game/images/offwhite-avatar-backer.png);
  background-size: cover;
  image-rendering: pixelated;
  border: 2px solid #4776bb;
  box-shadow: 1px 1px 1px#4776bb;
}

.student-dash-avatar {
  margin: auto;
  height: 60px;
  width: 33px;
  position: absolute;
  left: 0;
  right: 0;
  bottom: 11px;
}

.student-dash-avatar.one {
  background: url(../Game/pcs/char1/stand.png);
  image-rendering: pixelated;
  background-size: cover;
  background-repeat: no-repeat;
}

.student-dash-avatar.two {
  background: url(../Game/pcs/char2/stand.png);
  image-rendering: pixelated;
  background-size: cover;
  background-repeat: no-repeat;
}

.student-dash-avatar.three {
  background: url(../Game/pcs/char3/stand.png);
  image-rendering: pixelated;
  background-size: cover;
  background-repeat: no-repeat;
}

.student-dash-avatar.four {
  background: url(../Game/pcs/char4/stand.png);
  image-rendering: pixelated;
  background-size: cover;
  background-repeat: no-repeat;
}

.student-dash-avatar.five {
  background: url(../Game/pcs/char5/stand.png);
  image-rendering: pixelated;
  background-size: cover;
  background-repeat: no-repeat;
  width: 37px;
}

.student-dash-avatar.six {
  background: url(../Game/pcs/char6/stand.png);
  image-rendering: pixelated;
  background-size: cover;
  background-repeat: no-repeat;
  width: 37px;
}

.student-dash-avatar.nine {
  background: url(../Game/pcs/char9/stand.png);
  image-rendering: pixelated;
  background-size: cover;
  background-repeat: no-repeat;
  width: 36px;
  height: 60px;
}

.student-dash-avatar.ten {
  background: url(../Game/pcs/char10/stand.png);
  image-rendering: pixelated;
  background-size: cover;
  background-repeat: no-repeat;
  width: 42px;
  height: 60px;
}

.student-dash-avatar.seven {
  background: url(../Game/pcs/char7/stand.png);
  image-rendering: pixelated;
  background-size: cover;
  background-repeat: no-repeat;
  width: 42px;
  height: 62px;
}

.student-dash-avatar.eight {
  background: url(../Game/pcs/char8/stand.png);
  image-rendering: pixelated;
  background-size: cover;
  background-repeat: no-repeat;
  width: 43px;
  height: 62px;
}

.hearts-and-timer {
  display: grid;
  grid-template-columns: 1fr 300px 1fr;
  align-items: center;
  justify-content: center;
  width: 100%;
  height: 100px;
}

.student-name {
  grid-row: 1;
  height: 20px;
  text-align: start;
  margin-bottom: 0;
  margin-top: 3.2rem;
  margin-left: 2rem;
}

.student-class-link {
  grid-column: 2;
  text-align: start;
  margin-top: 0.5rem;
  margin-left: 2rem;
}

.onboarding-link {
  position: absolute;
  right: 0;
  top: 0;
  margin: 2.5rem 2.5rem;
  background: url(./images/help.png);
  width: 18px;
  height: 27px;
  background-size: cover;
  background-repeat: no-repeat;
  border: none;
  image-rendering: pixelated;
  transition: 0.1s;
}

.onboarding-link:hover {
  cursor: pointer;
  transform: scale(1.1);
}

.onboarding-link:focus {
  outline: none;
}

.onboarding-link:active {
  transform: scale(1);
}

.new-game-section {
  display: flex;
  position: relative;
  height: 40px;
}

.start-game {
  font-family: "Scriptorium";
  color: #4776bb;
  font-size: 2rem;
  background: #e3e8d9;
  text-shadow: 1px 1px 1px rgba(0, 0, 0.5);
  box-shadow: 1px 1px 1px#4776bb;
  margin-left: 3rem;
  height: 4rem;
  padding: 1rem;
}

.start-game:hover {
  cursor: pointer;
  box-shadow: 2px 2px 2px#4776bb;
}

.start-game:active {
  box-shadow: none;
}

.start-game:focus {
  outline: none;
}

.instrument-label {
  margin: 1.5rem 1rem 0 8rem;
}

.instrument-dropdown {
  background: #4776bb;
  width: 236px;
  height: 24px;
  line-height: 26px;
  margin: 1.2rem 0 0;
  padding-right: 2rem;
  transition: 0.2s;
  box-shadow: 1px 1px 1px#4776bb;
}

.instrument-dropdown:hover {
  cursor: pointer;
  box-shadow: 2px 2px 2px#000;
}

.instrument-dropdown:active {
  box-shadow: none;
}

.deploy-arrow {
  position: absolute;
  top: 26px;
  left: 702px;
  width: 14px;
  height: 10px;
  background: url(./images/arrow.png);
  background-size: cover;
  image-rendering: pixelated;
  transition: 0.3s;
  z-index: 30;
}

.deploy-arrow.true {
  transform: rotate(90deg);
}

.deploy-arrow:hover {
  cursor: pointer;
}

.instrument-dropdown-list {
  margin: 0;
  height: 0;
  overflow: hidden;
  list-style: none;
  padding: 0;
  position: absolute;
  left: 500px;
  top: 49px;
  padding: 0 1rem;
  background: #e3e8d9;
  transition: 0.3s;
  border: 0px solid #4776bb;
  z-index: 400;
}

.instrument-dropdown-list.true {
  height: 338px;
  padding-bottom: 1rem;
  box-shadow: 1px 1px 1px#4776bb;
  border: 2px solid #4776bb;
}

.instrument-list-item {
  margin-top: 0.4rem;
  width: 170px;
  font-size: 1.1rem;
}

.instrument-list-item:hover {
  cursor: pointer;
}

.total-games-played {
  text-align: start;
  margin: 5rem 0 2rem 3rem;
}

.most-recent-badges {
  margin-top: 2rem;
  text-align: start;
  margin-left: 3rem;
  margin-bottom: 0.5rem;
}

@keyframes snow {
  0% {
    top: -3000px;
    left: -3000px;
    opacity: 0;
  }
  1% {
    opacity: 1;
  }

  32% {
    opacity: 1;
  }
  34% {
    opacity: 0;
    top: 0;
    left: -1000px;
  }
  100% {
    opacity: 0;
    top: -3000px;
    left: -3000px;
  }
}

.fastest-times {
  text-align: start;
  margin: 3rem 0 0 3rem;
}

.times-row {
  display: flex;
  margin-left: 3rem;
}

.time-container {
  margin: 0.3rem 1rem 0 0;
}

.time-label {
  font-size: 1.3rem;
  margin-top: 0;
  text-align: start;
}

.badge-row {
  display: flex;
  margin-left: 3rem;
  background: #4776bb;
  width: 700px;
}

.badge {
  width: 39px;
  height: 33px;
  image-rendering: pixelated;
  background: url("./images/purple-gem.png");
  background-size: cover;
  margin: 1rem;
  position: relative;
}

.badge:hover {
  cursor: pointer;
}

.badge.level_one {
  background: url("./images/green-gem.png");
  background-size: cover;
}

.badge.level_one_perfect {
  background: url("./images/green-gem-stars.png");
  background-size: cover;
}

.badge.level_two {
  background: url("./images/blue-gem.png");
  background-size: cover;
}

.badge.level_two_perfect {
  background: url("./images/blue-gem-stars.png");
  background-size: cover;
}

.badge.level_three {
  background: url("./images/red-gem.png");
  background-size: cover;
}

.badge.level_three_perfect {
  background: url("./images/red-gem-stars.png");
  background-size: cover;
}

.badge.level_four {
  background: url("./images/gold-gem.png");
  background-size: cover;
}

.badge.level_four_perfect {
  background: url("./images/gold-gem-stars.png");
  background-size: cover;
}

.badge:hover .badge-descriptor {
  display: block;
}

.badge-descriptor {
  position: absolute;
  display: none;
  width: 100px;
  background: #4776bb;
  top: 20px;
  padding: 0.5rem;
  left: -40px;
}
