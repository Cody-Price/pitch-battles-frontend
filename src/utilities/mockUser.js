const mockUser = {
  first_name: "Chet",
  last_name: "Manly",
  avatar: 10,
  role: 0,
  email: "cmanly@gmail.com",
  games_played: 100,
  class: "6th-grade band"
};

export default mockUser;
