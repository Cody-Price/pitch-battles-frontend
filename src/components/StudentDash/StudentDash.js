import React, { Component } from "react";
import "./StudentDash.css";

import avatars from "../../utilities/avatars";
import { instruments } from "../../utilities/instruments";
import { timesHelper } from "../../utilities/timesHelper";
import { badgeHelper } from "../../utilities/badgeHelper";
import { fastestTimes } from "../../utilities/fastestTimes";

import LoadingAnimation from "../LoadingAnimation/LoadingAnimation";

class StudentDash extends Component {
  constructor() {
    super();

    this.state = {
      instrument: "choose your instrument...",
      dropdownDeploy: false,
      noInstrumentError: false
    };
  }

  handleInstrumentDropdown = () => {
    this.setState({
      dropdownDeploy: !this.state.dropdownDeploy
    });
  };

  selectInstrument = instrument => {
    this.setState({
      instrument,
      dropdownDeploy: false
    });
  };

  handleNewGame = () => {
    if (this.state.instrument === "choose your instrument...") {
      this.setState({ noInstrumentError: true });
      setTimeout(this.clearNoInstrumentError, 5000);
    } else {
      this.props.startGame(true, this.state.instrument);
    }
  };

  clearNoInstrumentError = () => {
    this.setState({ noInstrumentError: false });
  };

  render() {
    const instrumentList = instruments.map(instrument => {
      return (
        <li
          onClick={() => {
            this.selectInstrument(instrument.instrument);
          }}
          className={`instrument-list-item ${instrument.instrument}`}
          key={instrument.instrument}
        >
          {instrument.instrument}
        </li>
      );
    });

    let badges = this.props.user.attributes.badges.data.filter(badge => {
      return badge[Object.keys(badge)[0]];
    });

    if (badges.length > 10) {
      badges.sort((a, b) => {
        return b.id - a.id;
      });
      let newBadges = [];
      for (let i = 0; i < 10; i++) {
        newBadges.push(badges[i]);
      }
      badges = newBadges;
    }

    badges = badges.map((badge, index) => {
      const badgeClass = badgeHelper.find(helperBadge => {
        return helperBadge.rank == badge.id;
      });

      return (
        <div key={badge.attributes.name} className={`badge ${badgeClass.name}`}>
          <h4 className="badge-descriptor">{badge.attributes.name}</h4>
        </div>
      );
    });

    const displayTimes = fastestTimes(this.props.user);

    const times = displayTimes.map(time => {
      return (
        <div key={time.name} className={"time-container ${time.klass}"}>
          <h3 className="time-label">
            {time.name} - {time.time}
          </h3>
        </div>
      );
    });

    return (
      <main className="student-dash">
        {!this.props.user && <LoadingAnimation />}
        {this.props.user && (
          <section className="student-dash-wrapper">
            <header className="student-dash-header">
              <div
                className="avatar-circle"
                onClick={() => {
                  this.props.navigate("student account");
                }}
              >
                {" "}
                <h3 className="account-link-text">account page</h3>
              </div>
              <div className="student-dash-avatar-backer">
                <div
                  className={`student-dash-avatar ${
                    avatars[this.props.user.attributes.avatar]
                  }`}
                />
              </div>
              <section className="header-data">
                <h2 className="student-name">
                  {this.props.user.attributes.first_name}{" "}
                  {this.props.user.attributes.last_name}
                </h2>
                <h2 className="student-class-link">{this.props.user.class}</h2>
              </section>
              <button
                onClick={() => {
                  this.props.navigate("onboarding");
                }}
                className="onboarding-link"
              />
            </header>
            <section className="new-game-section">
              <button className="start-game" onClick={this.handleNewGame}>
                to battle!
              </button>
              <h2
                className={`no-instrument-warning ${
                  this.state.noInstrumentError
                }`}
              >
                you must select an instrument before playing...
              </h2>
              <h2 className="instrument-label">instrument:</h2>
              <h3
                onClick={this.handleInstrumentDropdown}
                className="instrument-dropdown"
              >
                {this.state.instrument}
              </h3>
              <div
                onClick={this.handleInstrumentDropdown}
                className={`deploy-arrow ${this.state.dropdownDeploy}`}
              />
              <ul
                className={`instrument-dropdown-list ${
                  this.state.dropdownDeploy
                }`}
              >
                {instrumentList}
              </ul>
            </section>
            <section className="student-dash-stats">
              <h2 className="total-games-played">
                total battles fought:{" "}
                <span>{this.props.user.attributes.total_games_played}</span>
              </h2>
              <h2 className="fastest-times">fastest times:</h2>
              <div className="times-row">{times}</div>
              <h2 className="most-recent-badges">top badges:</h2>
              <div className="badge-row">{badges}</div>
            </section>
          </section>
        )}
      </main>
    );
  }
}

export default StudentDash;
