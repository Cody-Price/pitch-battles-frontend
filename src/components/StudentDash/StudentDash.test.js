import React from "react";
import mockUser from "../../utilities/mockUser";

import StudentDash from "./StudentDash";
import Enzyme, { shallow } from "enzyme";
const EnzymeAdapter = require("enzyme-adapter-react-16");

Enzyme.configure({ adapter: new EnzymeAdapter() });

jest.mock("../../utilities/fetchCalls");

describe("StudentDash", () => {
  let wrapper;
  let mockToggleGame;
  let mockNavigate;

  beforeEach(() => {
    mockToggleGame = jest.fn();
    mockNavigate = jest.fn();

    wrapper = shallow(
      <StudentDash
        startGame={mockToggleGame}
        user={mockUser}
        navigate={mockNavigate}
      />
    );
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe("handleInstrumentDropdown", () => {
    it("should set state", () => {
      const expected = true;

      wrapper.instance().handleInstrumentDropdown();

      expect(wrapper.state().dropdownDeploy).toEqual(expected);
    });

    it("should call handleInstrumentDropdown on click", () => {
      const spy = jest.spyOn(wrapper.instance(), "handleInstrumentDropdown");
      wrapper.instance().forceUpdate();

      wrapper.find(".instrument-dropdown").simulate("click");

      expect(spy).toHaveBeenCalled();
    });
  });

  describe("selectInstrument", () => {
    it("should set state", () => {
      const expected = {
        instrument: "trumpet",
        dropdownDeploy: false,
        noInstrumentError: false,
        leaveClassWarning: false
      };

      wrapper.instance().selectInstrument("trumpet");

      expect(wrapper.state()).toEqual(expected);
    });

    it("should call selectInstrument on click", () => {
      const spy = jest.spyOn(wrapper.instance(), "selectInstrument");

      wrapper.find(".bassoon").simulate("click", "bassoon");

      expect(spy).toHaveBeenCalledWith("bassoon");
    });
  });

  describe("handleNewGame", () => {
    it("should set an error in state if no instrument is in state", () => {
      const expected = true;

      wrapper.instance().handleNewGame();

      expect(wrapper.state().noInstrumentError).toEqual(expected);
    });

    it("should call setTimeout with the correct params if no instrument is in state", () => {
      const spy = jest.spyOn(wrapper.instance(), "clearNoInstrumentError");
      jest.useFakeTimers();

      wrapper.instance().handleNewGame();

      expect(setTimeout).toHaveBeenCalledWith(spy, 5000);
    });

    it("should call startGame if there is an instrument in state with the correct params", () => {
      wrapper.setState({
        instrument: "flute"
      });
      const expected = "flute";

      wrapper.instance().handleNewGame();

      expect(mockToggleGame).toHaveBeenCalledWith(true, expected);
    });

    it("should call handleNewGame on click", () => {
      const spy = jest.spyOn(wrapper.instance(), "handleNewGame");
      wrapper.instance().forceUpdate();

      wrapper.find(".start-game").simulate("click");

      expect(spy).toHaveBeenCalled();
    });
  });

  describe("clearNoInstrumentError", () => {
    it("should call setState", () => {
      wrapper.setState({
        noInstrumentError: true
      });

      wrapper.instance().clearNoInstrumentError();

      expect(wrapper.state().noInstrumentError).toEqual(false);
    });
  });

  describe("navigate", () => {
    it("should call navigate with the appropriate params", () => {
      const expected = "onboarding";

      wrapper.find(".onboarding-link").simulate("click");

      expect(mockNavigate).toHaveBeenCalledWith(expected);
    });

    it("should call navigate with the appropriate params", () => {
      const expected = "student account";

      wrapper.find(".avatar-circle").simulate("click");

      expect(mockNavigate).toHaveBeenCalledWith(expected);
    });

    it("should call navigate with the appropriate params", () => {
      const expected = "student class view";

      wrapper.find(".student-class-link").simulate("click");

      expect(mockNavigate).toHaveBeenCalledWith(expected);
    });
  });
});
