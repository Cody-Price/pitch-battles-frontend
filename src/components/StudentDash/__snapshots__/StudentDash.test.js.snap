// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StudentDash should match the snapshot 1`] = `
<main
  className="student-dash"
>
  <section
    className="student-dash-wrapper"
  >
    <header
      className="student-dash-header"
    >
      <div
        className="student-dash-avatar-backer"
      >
        <div
          className="student-dash-avatar nine"
        />
      </div>
      <section
        className="header-data"
      >
        <h2
          className="student-name"
        >
          Chet
           
          Manly
        </h2>
        <h2
          className="student-class-link"
        >
          6th-grade band
        </h2>
      </section>
      <button
        className="onboarding-link"
        onClick={[Function]}
      />
    </header>
    <section
      className="new-game-section"
    >
      <button
        className="start-game"
        onClick={[Function]}
      >
        to battle!
      </button>
      <h2
        className="instrument-label"
      >
        instrument:
      </h2>
      <h3
        className="instrument-dropdown"
        onClick={[Function]}
      >
        choose your instrument...
      </h3>
      <div
        className="deploy-arrow false"
        onClick={[Function]}
      />
      <ul
        className="instrument-dropdown-list false"
      >
        <li
          className="instrument-list-item flute"
          key="flute"
          onClick={[Function]}
        >
          flute
        </li>
        <li
          className="instrument-list-item oboe"
          key="oboe"
          onClick={[Function]}
        >
          oboe
        </li>
        <li
          className="instrument-list-item clarinet"
          key="clarinet"
          onClick={[Function]}
        >
          clarinet
        </li>
        <li
          className="instrument-list-item alto saxophone"
          key="alto saxophone"
          onClick={[Function]}
        >
          alto saxophone
        </li>
        <li
          className="instrument-list-item tenor saxophone"
          key="tenor saxophone"
          onClick={[Function]}
        >
          tenor saxophone
        </li>
        <li
          className="instrument-list-item baritone saxophone"
          key="baritone saxophone"
          onClick={[Function]}
        >
          baritone saxophone
        </li>
        <li
          className="instrument-list-item bassoon"
          key="bassoon"
          onClick={[Function]}
        >
          bassoon
        </li>
        <li
          className="instrument-list-item trumpet"
          key="trumpet"
          onClick={[Function]}
        >
          trumpet
        </li>
        <li
          className="instrument-list-item horn"
          key="horn"
          onClick={[Function]}
        >
          horn
        </li>
        <li
          className="instrument-list-item trombone"
          key="trombone"
          onClick={[Function]}
        >
          trombone
        </li>
        <li
          className="instrument-list-item baritone"
          key="baritone"
          onClick={[Function]}
        >
          baritone
        </li>
        <li
          className="instrument-list-item violin"
          key="violin"
          onClick={[Function]}
        >
          violin
        </li>
        <li
          className="instrument-list-item viola"
          key="viola"
          onClick={[Function]}
        >
          viola
        </li>
        <li
          className="instrument-list-item cello"
          key="cello"
          onClick={[Function]}
        >
          cello
        </li>
        <li
          className="instrument-list-item string bass"
          key="string bass"
          onClick={[Function]}
        >
          string bass
        </li>
        <li
          className="instrument-list-item mallet percussion (high)"
          key="mallet percussion (high)"
          onClick={[Function]}
        >
          mallet percussion (high)
        </li>
        <li
          className="instrument-list-item mallet percussion (low)"
          key="mallet percussion (low)"
          onClick={[Function]}
        >
          mallet percussion (low)
        </li>
      </ul>
    </section>
    <section
      className="student-dash-stats"
    >
      <h2
        className="total-games-played"
      >
        total battles fought: 
        <span>
          115
        </span>
      </h2>
      <h2
        className="fastest-times"
      >
        fastest times:
      </h2>
      <div
        className="times-row"
      >
        <div
          className="time-container {$Objecy.keys(time)}"
          key="level_one"
        >
          <h3
            className="time-label"
          >
            level one
             - 
            11.01
          </h3>
        </div>
        <div
          className="time-container {$Objecy.keys(time)}"
          key="level_two"
        >
          <h3
            className="time-label"
          >
            level two
             - 
            15.24
          </h3>
        </div>
        <div
          className="time-container {$Objecy.keys(time)}"
          key="level_three"
        >
          <h3
            className="time-label"
          >
            level three
             - 
            24.64
          </h3>
        </div>
        <div
          className="time-container {$Objecy.keys(time)}"
          key="level_four"
        >
          <h3
            className="time-label"
          >
            level four
             - 
            30.17
          </h3>
        </div>
        <div
          className="time-container {$Objecy.keys(time)}"
          key="all"
        >
          <h3
            className="time-label"
          >
            all levels
             - 
            83.16
          </h3>
        </div>
      </div>
      <h2
        className="most-recent-badges"
      >
        most recent badges:
      </h2>
      <div
        className="badge-row"
      >
        <div
          className="badge one_hundred_games"
          key="one_hundred_games"
        >
          <h4
            className="badge-descriptor"
          >
            play 100 games
          </h4>
        </div>
        <div
          className="badge level_four"
          key="level_four"
        >
          <h4
            className="badge-descriptor"
          >
            level four: completed
          </h4>
        </div>
        <div
          className="badge fifty_games"
          key="fifty_games"
        >
          <h4
            className="badge-descriptor"
          >
            play 50 games
          </h4>
        </div>
        <div
          className="badge twenty_games"
          key="twenty_games"
        >
          <h4
            className="badge-descriptor"
          >
            play 20 games
          </h4>
        </div>
        <div
          className="badge level_three"
          key="level_three"
        >
          <h4
            className="badge-descriptor"
          >
            level three: completed
          </h4>
        </div>
        <div
          className="badge level_two_perfect"
          key="level_two_perfect"
        >
          <h4
            className="badge-descriptor"
          >
            level two: perfect
          </h4>
        </div>
        <div
          className="badge ten_games"
          key="ten_games"
        >
          <h4
            className="badge-descriptor"
          >
            play 10 games
          </h4>
        </div>
        <div
          className="badge level_two"
          key="level_two"
        >
          <h4
            className="badge-descriptor"
          >
            level two: completed
          </h4>
        </div>
        <div
          className="badge level_one_perfect"
          key="level_one_perfect"
        >
          <h4
            className="badge-descriptor"
          >
            level one: perfect
          </h4>
        </div>
        <div
          className="badge level_one"
          key="level_one"
        >
          <h4
            className="badge-descriptor"
          >
            level one: completed
          </h4>
        </div>
      </div>
    </section>
  </section>
</main>
`;
